### PR TITLE
[keycloak] Add service account creation

### PR DESCRIPTION
**Pull request by denniseffing**
This PR adds service account creation to the Keycloak helm chart.

Service account creation can be configured by setting `keycloak.serviceAccount.create` and `keycloak.serviceAccount.name`.

### CODE REVIEW

File: charts/keycloak/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: {{ template "keycloak.name" . }}
+    chart: {{ template "keycloak.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: {{ template "keycloak.fullname" . }}
+{{- end -}}
```
Comment (unguiculus) [inline charts/keycloak/templates/serviceaccount.yaml]: Add line break at end of file.

File: charts/keycloak/templates/statefulset.yaml [hunk 2]
```diff
@@ -38,6 +38,9 @@ spec:
 {{ toYaml . | indent 8 }}
 {{- end }}
       enableServiceLinks: {{ .Values.keycloak.enableServiceLinks }}
+{{- if or (.Values.keycloak.serviceAccount.create) (.Values.keycloak.serviceAccount.name) }}
```
Comment (unguiculus) [inline charts/keycloak/templates/statefulset.yaml]: The if statement is unnecessary. Please remove it.

File: charts/keycloak/values.yaml [hunk 3]
```diff
@@ -28,6 +28,13 @@ keycloak:
 
   enableServiceLinks: false
 
+  serviceAccount:
+    # Specifies whether a service account should be created
+    create: true
```
Comment (unguiculus) [inline charts/keycloak/values.yaml]: I'd set it to `false` by default.

**Comment by denniseffing** [comment]
I added the requested changes. CI seems to complain because it wants me to bump the version number again, however imo that doesn't make much sense.

**Pull request merged**